improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,578 @@
+
+    Args:
+        interpreter (MOMetadataInterpreter):
+            Populated instance of MOMetadataInterpreter
+        verbose (bool):
+            Optional flag to include information about the source of the
+            metadata interpretation (eg name, coordinates, attributes, etc)        
+
+    Returns:
+        str:
+            Formatted string describing metadata in human-readable form
+    """
+    if interpreter.unhandled:
+        return f"{interpreter.diagnostic} is not handled by this interpreter"
+
+    def vstring(source_metadata):
+        """Format additional message for verbose output"""
+        return f"    Source: {source_metadata}"
+
+    def format_standard_cases(interpreter):
+        """Format prob / perc / diagnostic information"""
+        field_type = interpreter.field_type.replace("_", " ")
+        diagnostic = interpreter.diagnostic.replace("_", " ")
+        if interpreter.relative_to_threshold:
+            relative_to_threshold = interpreter.relative_to_threshold.replace("_", " ")
+
+        rval = []
+        rtt = (
+            f" {relative_to_threshold} thresholds"
+            if interpreter.field_type == interpreter.PROB
+            else ""
+        )
+        rval.append(f"It contains {field_type} of {diagnostic}{rtt}")
+        if verbose:
+            rval.append(vstring("name, threshold coordinate (probabilities only)"))
+
+        if interpreter.methods:
+            rval.append(f"These {field_type} are of {diagnostic}{interpreter.methods}")
+            if verbose:
+                rval.append(vstring("cell methods"))
+
+        ppstring = "some" if interpreter.post_processed else "no"
+        rval.append(f"It has undergone {ppstring} significant post-processing")
+        if verbose:
+            rval.append(vstring("title attribute"))
+        return rval
+
+    field_type = interpreter.field_type.replace("_", " ")
+    output = []
+    output.append(f"This is a {interpreter.prod_type} {field_type} file")
+    if verbose:
+        output.append(vstring("name, coordinates"))
+
+    if interpreter.diagnostic not in SPECIAL_CASES:
+        output.extend(format_standard_cases(interpreter))
+
+    if interpreter.diagnostic in ANCILLARIES:
+        output.append(f"This is a static ancillary with no time information")
+    elif interpreter.blended:
+        output.append(f"It contains blended data from models: {interpreter.model}")
+        if verbose:
+            output.append(vstring("title attribute, model ID attribute"))
+    else:
+        if interpreter.model:
+            output.append(f"It contains data from {interpreter.model}")
+            if verbose:
+                output.append(vstring("model ID attribute"))
+        else:
+            output.append("It has no source model information and cannot be blended")
+            if verbose:
+                output.append(vstring("model ID attribute (missing)"))
+
+    if interpreter.warnings:
```

Comment:
Why are you choosing to raise the errors in the run method and the warnings here?

Reply:
The errors need to be raised by the interpreter, because they are saying that the file is essentially invalid.  By contrast these warnings aren't so much code warnings (like our warnings on things like "not tested with masked input"), as additional guidance for the developer that goes with the interpretation of the file.  I want them to be read in context with the other information that's been given from the interpreter.

Reply:
Ok, that makes sense